### DISCUSSION TITLE
Why does dracut not add leading and trailing spaces?

### DISCUSSION THREAD

**Discussion by ChristophSchmidpeter** [dracutdevs/dracut#1980] [Q&A]
According to [Gentoo's Dracut wiki](https://wiki.gentoo.org/wiki/Dracut#Adding_modules), within Dracut .conf files, assignments of parameter (lists) like `add_dracutmodules` must contain leading and trailing spaces. E.g.
`add_dracutmodules+=" module " # Note leading and trailing spaces`

Assuming this is correct, are there cases where one intentionally should/wants to omit leading and/or trailing spaces? If not, wouldn't it make sense to have Dracut automatically insert leading and trailing spaces where appropriate for improved convenience and to avoid a potential source of (user) error?

**Comment by dtardon** [comment]
This is not possible to do with the current way of processing of configuration files. The problem is that the "configuration" files are not really configuration--they are shell scripts which are sourced into dracut's shell process. IOW, from dracut's POV, each config. file is an opaque unit. The best dracut could do is to append a space to all the "list" variables (`add_dracutmodules` etc.) after reading each config. file, but that wouldn't help if the same variable is used multiple times in a config. file...
